Custom Field Constraints/CFConstraints.data.provider.js

```diff
@@ -98,6 +100,38 @@ tau.mashups
                 return getTasksDeferred.promise();
             },
 
+            getTeamProjectsPromise: function(entitiesDetailed) {
+                var getTeamProjectsDetails = $.Deferred();
+                var teamProjectIds = _.chain(entitiesDetailed)
+                    .filter(function(entityDetailed) {
+                        return entityDetailed.project &&
+                            entityDetailed.assignedTeams && entityDetailed.assignedTeams.length > 0;
+                    })
+                    .map(function(entityDetailed) {
+                        return _.pluck(entityDetailed.project.teamProjects, 'id');
+                    })
+                    .flatten()
+                    .unique()
+                    .value();
+                if (teamProjectIds.length === 0) {
+                    getTeamProjectsDetails.resolve([]);
+                    return getTeamProjectsDetails.promise();

```

Comment:
You could use `return $.when([]);` here and define  `var getTeamProjectsDetails = $.Deferred();` before the usage